packages/noports_core/lib/src/sshnp/impl/notification_request_message.dart

```diff
@@ -1,33 +1,29 @@
-abstract class SshnpSessionRequest {
-  late bool direct;
-  late String sessionId;
-  late String host;
-  late int port;
+class SshnpSessionRequest {
+  final bool direct;
+  final String sessionId;
+  final String host;
+  final int port;
+  final bool authenticateToRvd;
+  final String clientNonce;
+  final String rvdNonce;
 
-  Map message();
-}
-
-class SessionIdMessage extends SshnpSessionRequest{
-
-  @override
-  Map message() {
-    Map m = {};
-    m['direct'] = true;
-    m['sessionId'] = sessionId;
-    m['host'] = host;
-    m['port'] = port;
+  SshnpSessionRequest({
+    required this.direct,
+    required this.sessionId,
+    required this.host,
+    required this.port,
+    required this.authenticateToRvd,
+    required this.clientNonce,
+    required this.rvdNonce,
+  });
 
-    return m;
-  }
+  Map<String, dynamic> toJson() => {
+        'direct': direct,
+        'sessionId': sessionId,
+        'host': host,
+        'port': port,
+        'authenticateToRvd': authenticateToRvd,
+        'clientNonce': clientNonce,
+        'rvdNonce': rvdNonce,
+      };
```

Comment:
Going forward, we might want to consider a lightweight JSON spec for these various payloads - considering we have a Dart & Python impl, with C on the way.

Reply:
This might cause more tech debt, without any real benefit though...